cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -202,11 +222,28 @@ func GetIntermediates() (*x509.CertPool, error) {
 	return fulcioroots.GetIntermediates()
 }
 
-func NewClient(fulcioURL string) (api.LegacyClient, error) {
+func NewClient(fulcioURL string) (fulciopb.CAClient, error) {
+	var opts []grpc.DialOption
 	fulcioServer, err := url.Parse(fulcioURL)
 	if err != nil {
 		return nil, err
 	}
-	fClient := api.NewClient(fulcioServer, api.WithUserAgent(options.UserAgent()))
-	return fClient, nil
+	host := fulcioServer.Host
+	switch fulcioServer.Scheme {
+	case "https":
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{MinVersion: tls.VersionTLS12})))
+		if fulcioServer.Port() == "" {
+			host = fmt.Sprintf("%s:443", host)
+		}
+	default:
```

Comment:
There's an open issue to add an insecure flag - https://github.com/sigstore/cosign/issues/2290

We should only default to a TLS-less connection with that flag set. Revising the above comment, if we match on port and or scheme:

* scheme=https => TLS
* port=443 => TLS
* insecure flag set => no TLS

We could just ignore port 80, and in testing it's unlikely you'd be using port 80 anyways, probably 8000+